feat(docs): tips and tricks (#543)
catppuccin/userstyles: docs/tips-and-tricks.md

You may use the [Flow Circular](https://fonts.google.com/specimen/Flow+Circular) font to redact details by obscuring any text on the page.

Use this snippet at the top of your userstyle to import and use the font:

```less
@import url('https://fonts.googleapis.com/css2?family=Flow+Circular&display=swap');

* {
  font-family: 'Flow Circular', cursive;
}
```

### Converting images to WebP

The Catppuccin project prefers to use the [WebP image format](https://en.wikipedia.org/wiki/WebP) for the asset preview images. We recommend using Google's [`cwebp`](https://developers.google.com/speed/webp/docs/cwebp) conversion utility to convert images to WebP.

#### Installation

| Method                                | Instructions                                                                           |
| ------------------------------------- | -------------------------------------------------------------------------------------- |
| [Homebrew](https://brew.sh/)          | `brew install webp` [[view cask]](https://formulae.brew.sh/formula/webp)               |
| [MacPorts](https://www.macports.org/) | `sudo port install webp` [[view port]](https://ports.macports.org/port/webp/)          |

> | Binaries                              | See https://storage.googleapis.com/downloads.webmproject.org/releases/webp/index.html. |

Author: I think we should just push users into some other guide to install that's more official, since it will cover most systems.

Reviewer: There isn't a more official guide. https://developers.google.com/speed/webp/docs/precompiled lists exactly what I have in the table above: Homebrew, Macports, and binaries.

Author: Fair enough.

#### Usage

> > _See [the full documentation on `cwebp`](https://developers.google.com/speed/webp/docs/cwebp) for further reference._

Author: we could use the github flavoured tips here like we did for the other further reading before.

<p align="center">
  <h2 align="center">📖 Tips and tricks for writing a userstyle</h2>
</p>

> &nbsp;

Author: We could, I only removed it beacuse I couldn't think of anything.

### How can I see my changes in real time?

If you want to see your changes in real-time while using an external editor:

> 1. Head to `chrome://extensions/` or your browser's equivalent.

Reviewer: We could update https://github.com/catppuccin/userstyles/blob/main/docs/browsers.md and link to?

Author: This makes sense.

<p align="center">
  <h2 align="center">📖 Tips and Tricks for writing a userstyle</h2>
</p>

&nbsp;

### Live reloading

> If you want to see your changes in real-time and not have to use the stylus editor. This is the tip for you.

Reviewer 2: ```suggestion
If you want to see your changes in real-time while using an external editor:
```

1. Head to `chrome://extensions/` or your browser's equivalent.
2. Locate the stylus extension and click on the `Details` button.
3. Enable `Allow access to file URLs`.
4. Now you can open the `catppuccin.user.css` file in your browser by typing `file:///path/to/catppuccin.user.css` in the address bar.
5. Then you should notice stylus will load on that page, at this point ensure `live preview` is enabled.
6. Now you can make changes in your prefered editor and see real-time changes.

> ### Hideing sensitive information in screenshots

Reviewer 2: should be `Hiding`

When taking screenshots of your userstyle, you may want to hide sensitive information such as your username, email, or any other personal information.

> To do this we use the [flow circular font](https://fonts.google.com/specimen/Flow+Circular). 

Reviewer 2: probably capitalize it as `Flow Circular` consistently

<p align="center">

>   <h2 align="center">📖 Tips and Tricks for writing a userstyle</h2>

Reviewer: Capitalization is weird here.

</p>

&nbsp;

### Live reloading

If you want to see your changes in real-time while using an external editor:

1. Head to `chrome://extensions/` or your browser's equivalent.
2. Locate the stylus extension and click on the `Details` button.
3. Enable `Allow access to file URLs`.
4. Now you can open the `catppuccin.user.css` file in your browser by typing `file:///path/to/catppuccin.user.css` in the address bar.

> 5. Then you should notice stylus will load on that page, at this point ensure `live preview` is enabled.

Reviewer: ```suggestion
5. Stylus will load on that page, and at this point ensure that `live preview` is enabled.
```

6. Now you can make changes in your preferred editor and see real-time changes.

> [!TIP]
> Further reading: [Stylus Wiki (Initial installation and live reload)](https://github.com/openstyles/stylus/wiki/Writing-UserCSS#initial-installation-and-live-reload)

### Hiding sensitive information in screenshots

When taking screenshots of userstyles, you may want to hide sensitive information such as your username, email, or other personal information.

To do this, you we usually use the [Flow Circular](https://fonts.google.com/specimen/Flow+Circular) font to redact the details.

Use this snippet at the top of your userstyle to import and use the font:

```less
@import url('https://fonts.googleapis.com/css2?family=Flow+Circular&display=swap');

* {
  font-family: 'Flow Circular', cursive;
}

> ```

Reviewer: Hmm. Personally I have a separate userstyle that runs on all pages that I can toggle on and off separately:

```less
/* ==UserStyle==
@name           Flow Circular Font
@namespace      flow-circular
@version        1.0.0
@author         uncenter
==/UserStyle== */
@-moz-document regex("*") {
    @import url('https://fonts.googleapis.com/css2?family=Flow+Circular&display=swap');

    * {
        font-family: 'Flow Circular', cursive;
    }
}
```

Author: Thats fair but i think its fine to leave this as is, users can do whatever they think works.
